tests/unit/anchore_engine/analyzers/syft/test_syft.py

```diff
@@ -0,0 +1,130 @@
+                    {
+                        "parent": "parent-id",
+                        "child": "child-id",
+                        "type": "ownership-by-files",
+                    }
+                ],
+                ["parent-pkg"],
+            ),
+            (
+                [
+                    {
+                        "parent": "parent-id",
+                        "child": "child-id",
+                        "type": "ownership-by-files",
+                    },
+                    {
+                        "parent": "another-parent-id",
+                        "child": "child-id",
+                        "type": "ownership-by-files",
+                    },
+                ],
+                ["parent-pkg"],
+            ),
+            (
+                [
+                    {
+                        "parent": "UNCORRELATED-id",
+                        "child": "child-id",
+                        "type": "ownership-by-files",
+                    }
+                ],
+                ["parent-pkg"],
+            ),
+            #### filtering NOT applied
+            (
+                [
+                    {
+                        "parent": "parent-id",
+                        "child": "child-id",
+                        "type": "NOT-ownership-by-files",
+                    }
+                ],
+                ["child-pkg", "parent-pkg"],
+            ),
+            (
+                [
+                    {
+                        "parent": "parent-id",
+                        "child": "NOT-child-id",
+                        "type": "ownership-by-files",
+                    }
+                ],
+                ["child-pkg", "parent-pkg"],
+            ),
+            ([], ["child-pkg", "parent-pkg"]),
+        ],
+    )
+    def test_filter_artifact_by_relationships(
+        self, relationships, expected_artifact_names
+    ):
+        artifacts = [
+            {
+                "id": "child-id",
+                "name": "child-pkg",
+                "type": "rpm",
+            },
+            {
+                "id": "parent-id",
+                "name": "parent-pkg",
+                "type": "rpm",
+            },
+        ]
+
+        actual = list(filter_artifacts(artifacts, relationships))
+        assert [a["name"] for a in actual] == expected_artifact_names
+
+    def test_filter_artifact_missing_id(self):
+        # we are ensuring that filtering is not applied if IDs & relationships are missing
+        initial_artifacts = [
+            {
+                "name": "child-pkg",
+                "type": "rpm",
+            },
+            {
+                "name": "parent-pkg",
+                "type": "rpm",
+            },
+        ]
+
+        actual = list(filter_artifacts(initial_artifacts, []))
+        assert actual == initial_artifacts
+
+    @pytest.mark.parametrize(
+        "pkg_type,expected_artifact_names",
+        [
+            #### filtering applied
```

Comment:
this would be easier to grasp if these conditions were separate,  in a new test instead of inserting comments within the parametrization